### PR TITLE
[RLlib] Single agent RLTrainer made easy

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?

The `compute_loss` method is built for multi-agent. While it is still possible to write single-agent losses, it may become confusing to users. We should find a way to allow them to specify single-agent losses as well, without having to think about one extra layer of hierarchy for module ids. This PR makes it easy to this. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/rl_trainer/rl_trainer.py [hunk 1]
```diff
@@ -178,6 +177,48 @@ def compute_loss(
...
+        results_all_modules = {}
+        for module_id in fwd_out:
+            module_batch = batch[module_id]
+            module_fwd_out = fwd_out[module_id]
+
+            module_results = self._compute_loss_per_module(
+                module_id, module_batch, module_fwd_out
+            )
+            results_all_modules[module_id] = module_results
+            loss = module_results[self.TOTAL_LOSS_KEY]
```
Comment (gjoliver) [inline rllib/core/rl_trainer/rl_trainer.py:190]: just a bit confusing, why would module_results ever contain TOTAL_LOSS_KEY at this point?

File: rllib/core/testing/tf/bc_rl_trainer.py [hunk 2]
```diff
@@ -3,28 +3,17 @@
...
-                loss_total = loss
-            else:
-                loss_total += loss
-
-        loss_dict[self.TOTAL_LOSS_KEY] = loss_total
+        action_dist = fwd_out["action_dist"]
+        loss = -tf.math.reduce_mean(action_dist.log_prob(batch[SampleBatch.ACTIONS]))
 
-        return loss_dict
+        return {self.TOTAL_LOSS_KEY: loss}
```
Comment (gjoliver) [inline rllib/core/testing/tf/bc_rl_trainer.py:19]: maybe this should get saved under per-module key, so that rl_trainer can always aggregate the loss into a total key.
also, why is TOTAL_LOSS_KEY a contains on self?

**Review by gjoliver — COMMENTED** [review]
a couple of random questions.

File: rllib/core/rl_trainer/rl_trainer.py (hunk 1 above)
Comment (kouroshHakha) [reply rllib/core/rl_trainer/rl_trainer.py:190]: that's the convention. User needs to return their loss under that key.

Comment (kouroshHakha) [reply rllib/core/rl_trainer/rl_trainer.py:190]: they can optionally include other metrics they want to log.

File: rllib/core/testing/tf/bc_rl_trainer.py (hunk 2 above)
Comment (kouroshHakha) [reply rllib/core/testing/tf/bc_rl_trainer.py:19]: to be accessible everywhere. No need to import :)

Comment (kouroshHakha) [reply rllib/core/testing/tf/bc_rl_trainer.py:19]: > maybe this should get saved under per-module key, so that rl_trainer can always aggregate the loss into a total key.

This is more general. Trust me :)

Comment (gjoliver) [reply rllib/core/testing/tf/bc_rl_trainer.py:19]: ok :)

**Review by gjoliver — APPROVED** [review]
